test/copy.jl

```diff
@@ -108,3 +108,25 @@ let x = BigFloat[1:1000;], y, z, v
     # Check that the setprecision indeed does something
     @test z != x
 end
+
+# issue #19921
+type Foo
+    a::String
+end
+
+type Bar
+    foo::Foo
+    fooDict::Dict{Foo, Int64}
+end
```

Comment:
The type definitions should be able to go in the `@testset`. That would avoid a bit of scope clutter. Probably not a big deal though.